static/js/components/util/Prompt.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect } from "react"
+import { Prompt as ReactRouterPromp, PromptProps } from "react-router"
+
+interface Props extends PromptProps {
+  onBeforeUnload: boolean
+  message: string
+}
+
+/**
+ * A wrapper around ReactRouter's <Prompt /> component which also prompts for
+ * confirmation on beforeunload events (if `props.onBeforeUnload` is true).
+ */
+export default function Prompt(props: Props) {
+  const { onBeforeUnload, ...otherProps } = props
+  const { when, message } = otherProps
+  useEffect(() => {
+    if (!onBeforeUnload || !when) return
+    const listener = (event: BeforeUnloadEvent) => {
+      /**
+       * Most browsers won't show this, but Chrome, at least, requires some
+       * returnValue to be set.
+       */
+      event.returnValue = message
+      event.preventDefault()
+    }
+    window.addEventListener("beforeunload", listener, { capture: true })
+    const remove = () =>
+      window.removeEventListener("beforeunload", listener, { capture: true })
+    return remove
+  }, [onBeforeUnload, when, message])
+
+  return <ReactRouterPromp {...otherProps} />
```

Comment:
One consequence of using react-router's <Prompt /> is that confirmation is now requested via `window.confirm`, ie the browser's API. So the confirmation modal is now the browser's, not our customized version.

**Before**:
<img width="1612" alt="Screen Shot 2022-04-20 at 2 22 06 PM" src="https://user-images.githubusercontent.com/9010790/164297404-085fb16a-d5fb-4915-9017-52309c2dde51.png">
**After**
<img width="1303" alt="Screen Shot 2022-04-20 at 2 23 07 PM" src="https://user-images.githubusercontent.com/9010790/164297446-88ea3384-3e2e-467c-9a4a-c64a214b2cce.png">

IMO the old one looks slightly better, but using <Prompt /> is more canonical react router approach and lets use do this more easily without custom code. And we can customize the confirmation dialog used by react-router if we want, but it is not currently a priority.